File: j2/perm1.R
tic()
n = 1000

dup = 10^5 #10^5

rint = function(n1){
ceiling(runif(1)*n1)
}

#allv = rep(rep(-1, n),dup)
allv = matrix(list(), nrow=dup, ncol=n)
for (j in 1:dup) {
nv = 1:n
nend = n
for (i in 1:n) {
si = rint(nend)
allv[j,i] = nv[si]
nv[si] = nv[nend]
nend = nend - 1
}
#v
}
toc()
#allv

File: j2/perm1.py
from pytictoc import TicToc
t = TicToc()
t.tic()
from numpy.random import *
import numpy as np

n = 1000

dup = 10**5

def rint(n1):
	return randint(0,n1,size=(dup))

nm = np.zeros((dup,n),dtype=int)
row_idxs = np.arange(len(nm))

for i in range(n):
	si = rint(i+1)
	# for j in range(dup):
	# 	nm[j,i] = nm[j,si[j]]
	# 	nm[j,si[j]] = i+1
	nm[:,i] = nm[row_idxs,si]
	nm[row_idxs, si] = i+1	

t.toc()
print(nm)
a=np.sum(nm,axis=0)/10**5
print(a)

import seaborn as sns
import matplotlib.pyplot as plt
sns.distplot(a)
plt.show()